test/gst-msdk/transcode/transcoder.py

```diff
@@ -161,23 +168,27 @@ def transcode(self):
       " frame-checksum=false plane-checksum=false dump-output=true"
       " dump-location={srcyuv}".format(**vars(self)))
 
-    yuv = get_media()._test_artifact(
-      "{}.yuv".format(self.case))
-    refyuv_decode_arg = self.gen_refyuv_decode_commands(self.outputs[0]["codec"], "hw")
-    call(
-      "gst-launch-1.0 -vf filesrc location={}"
-      " ! {}"
-      " ! videoconvert ! video/x-raw,format=I420"
-      " ! checksumsink2 file-checksum=false qos=false"
-      " frame-checksum=false plane-checksum=false dump-output=true"
-      " dump-location={}".format(self.ofile, refyuv_decode_arg, yuv))
-    self.check_metrics(yuv)
-
-  def check_metrics(self, yuv):
+    for n, output in enumerate(self.outputs):
+      for channel in xrange(output.get("channels", 1)):
+        encoded = self.goutputs[n][channel]
+        yuv = get_media()._test_artifact(
+          "{}_{}_{}.yuv".format(self.case, n, channel))
+        refyuv_decode_arg = self.gen_refyuv_decode_commands(self.outputs[0]["codec"], "hw")
```

Comment:
`self.outputs[0]` is wrong.  Each output might use a different codec but you are using the codec from the first output for all outputs.  For example, `self.output[0]` might be `avc` and `self.output[1]` might be `mjpeg`.  You need to use the codec in current loop iteration of `output`.

Reply:
sorry, have fixed it.